### PR TITLE
Added Matrix Script in Python

### PR DESCRIPTION
**Pull request by mehabhalodiya**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Added Matrix Script in Python(HackerRank).

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Solves #1581

### CODE REVIEW

**Comment by mehabhalodiya** [comment]
Please check this @HarshCasper

File: Miscellaneous/HackerRank/MatrixScript.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+first_multiple_input = input().rstrip().split()
```
Comment (atarax665) [inline Miscellaneous/HackerRank/MatrixScript.py:26]: Add proper description of the problem in commented form along with the link of the hackerrank question.

File: Miscellaneous/HackerRank/MatrixScript.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+n = int(first_multiple_input[0])
+
+m = int(first_multiple_input[1])
+
+matrix = []
+
```
Comment (atarax665) [inline Miscellaneous/HackerRank/MatrixScript.py:33]: You may put the logic in a function and only have the input and output function in the main code. Also add proper comments to explain your approach.

File: Miscellaneous/HackerRank/MatrixScript.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    matrix_item = input()
+    matrix.append(matrix_item)
+string = ""
+for y in range(m):
+    for x in range(n):
+        string += matrix[x][y]
+pattern = re.compile('(\w)(\W)+(\w)')
+subs = pattern.sub(r"\1 \3", string)
+print(subs) 
+
```
Comment (atarax665) [inline Miscellaneous/HackerRank/MatrixScript.py:64]: Add sample input/output at the end of the program in commented form.

**Review by atarax665 — CHANGES_REQUESTED** [review]
You should have a look on our [Python coding template](https://github.com/TesseractCoding/NeoAlgo/blob/master/Python/Code_Template.md) and format your code accordingly.

**Comment by mehabhalodiya** [comment]
Kindly check it out @atarax665 .

File: Miscellaneous/HackerRank/MatrixScript.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+faster reuse. Regarding the specific regex expression, this searches for groups
+that have alphanumerics (that's the \w part) or apostrophes (which is also in
+those square brackets) that are 1 or longer. Note that whitespace is not a
+match, so this, generally speaking, breaks a line into words.
+
+Time Complexity: O(n^2)
+Space Complexity: O(n)
+'''
+
+import math
```
Comment (atarax665) [inline Miscellaneous/HackerRank/MatrixScript.py:20]: Do we really need to import all these in this program?

**Review by atarax665 — CHANGES_REQUESTED** [review]
Please update the README.md file.

File: Miscellaneous/HackerRank/MatrixScript.py (hunk 4 above)
Comment (mehabhalodiya) [reply Miscellaneous/HackerRank/MatrixScript.py:20]: Yes, HackerRank provides us all required libraries.

Comment (atarax665) [reply Miscellaneous/HackerRank/MatrixScript.py:20]: Its not needed atleast for this program, but anyways you can leave it as it is.

File: Miscellaneous/HackerRank/MatrixScript.py (hunk 2 above)
Comment (atarax665) [reply Miscellaneous/HackerRank/MatrixScript.py:33]: Add appropriate comments and then this PR will be good to merge.

File: Miscellaneous/HackerRank/MatrixScript.py [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+Explanation:
+
+The decoded script is:
+This$#is% Matrix#  %!
+It replaces the symbols or spaces between two alphanumeric characters with a
+single space ' ' for better readability.
+So, the final decoded script is:
+This is Matrix#  %!
+'''
+
```
Comment (atarax665) [inline Miscellaneous/HackerRank/MatrixScript.py]: Remove this extra blank line.

**Review by atarax665 — CHANGES_REQUESTED** [review]

**Review by atarax665 — APPROVED** [review]

**Comment by atarax665** [comment]
@all-contributors please add @mehabhalodiya for code

**Pull request merged**